services/director-v2/src/simcore_service_director_v2/cli/core.py

```diff
@@ -0,0 +1,298 @@
+        reraise=True,
+    ):
+        with attempt:
+            typer.echo(f"Attempting to save {node_uuid} {label}")
+            await dynamic_sidecar_client.save_service_state(
+                _get_dynamic_sidecar_endpoint(app.state.settings, node_uuid)
+            )
+
+
+async def async_project_save_state(project_id: ProjectID, retry_save: int) -> None:
+    async with _initialized_app() as app:
+        projects_repository: ProjectsRepository = get_repository(
+            app, ProjectsRepository
+        )
+        project_at_db = await projects_repository.get_project(project_id)
+
+        typer.echo(f"Saving project '{project_at_db.uuid}' - '{project_at_db.name}'")
+
+        dynamic_sidecar_client = api_client.get_dynamic_sidecar_client(app)
+        nodes_failed_to_save: list[NodeIDStr] = []
+        for node_uuid, node_content in project_at_db.workbench.items():
+            # onl dynamic-sidecars are used
+            if not await requires_dynamic_sidecar(
+                service_key=node_content.key,
+                service_version=node_content.version,
+                director_v0_client=DirectorV0Client.instance(app),
+            ):
+                continue
+
+            try:
+                await _save_node_state(
+                    app,
+                    dynamic_sidecar_client,
+                    retry_save,
+                    node_uuid,
+                    node_content.label,
+                )
+            except Exception:  # pylint: disable=broad-except
+                nodes_failed_to_save.append(node_uuid)
+
+    if nodes_failed_to_save:
+        typer.echo(
+            "The following nodes failed to save:"
+            + "\n- "
+            + "\n- ".join(nodes_failed_to_save)
+            + "\nPlease try to save them individually!"
+        )
+        sys.exit(1)
+
+    typer.echo(f"Save complete for project {project_id}")
+
+
+### NODE SAVE STATE
+
+
+async def async_node_save_state(node_id: NodeID, retry_save: int) -> None:
+    async with _initialized_app() as app:
+        dynamic_sidecar_client = api_client.get_dynamic_sidecar_client(app)
+        await _save_node_state(
+            app, dynamic_sidecar_client, retry_save, NodeIDStr(f"{node_id}")
+        )
+
+    typer.echo(f"Node {node_id} save completed")
+
+
+### PROJECT STATE
+
+
+class ServiceType(str, Enum):
+    COMP = "comp"
+    DYNAMIC = "dynamic"
+    FRONTEND = "frontend"
+
+
+class StatusIcon(str, Enum):
+    OK = ":green_heart:"
+    ONGOING = ":yellow_heart:"
+    FAILED = ":broken_heart:"
+    UNKNOWN = ":grey_question:"
+    HTTP_ERROR = ":exclamation:"
+
+
+class RenderData(BaseModel):
+    node_uuid: NodeIDStr
+    label: str
+    status_icon: StatusIcon
+    state: str
+
+
+def _get_service_type(service_key: str) -> ServiceType:
+    decoded_service_key = urllib.parse.unquote_plus(service_key)
+    service_type = decoded_service_key.split("/")[2]
+    return ServiceType(service_type)
```

Comment:
same for this fct. We do have at least one elsewhere. Please re-use or move to a common location. thanks!

Reply:
Similar but not the same. But I will put it in a shared location since it can be used by others and now relies on the models_library `ServiceType`